openedx/openedx-platform
# fix: missing and broken waffle flag links

## Description

The current header with `contentstore.new_studio_mfe.use_new_advanced_settings_page` enabled, the link leads to the a page not found or doesn't change pages. Now the function that generates the link is called and given the correct variable. This change impacts Authors.

## Testing instructions

1. Using Django admin, enable `contentstore.new_studio_mfe.use_new_advanced_settings_page`.
2. Reload Studio, use the nav bar and click on the Advanced Settings page.
3. Should be navigated to the page in the course-authoring MFE.
4. Navigate back to studio.
5. In the nav bar, click on another page that is NOT enabled by the flags listed above.
6. Should navigate to the page and remain in Studio.

## Deadline

None
